include/Gaffer/ShufflePlug.inl

```diff
@@ -77,7 +77,7 @@ T ShufflesPlug::shuffle( const T &sourceContainer ) const
 
 	NameContainer names;
 
-	for( const ConstShufflePlugPtr &plug : ShufflePlug::Range( *this ) )
+	for( const ConstShufflePlugPtr plug : ShufflePlug::Range( *this ) )
```

Comment:
Ugh! This relates to a mistake I made way back, which is documented in [`FilteredChildIterator.h`](https://github.com/GafferHQ/gaffer/blob/main/include/Gaffer/FilteredChildIterator.h#L67-L71) :

```
/// \todo It's inconvenient that our reference type
/// is ChildType::Ptr rather than just ChildType. It
/// leads to lots of ugly `it->get()` and `(*it)->`
/// calls. Change this for this class and also for
/// the RecursiveIterator classes.
````

From the error I get here, it looks like we either have the choice of `const ShufflePlugPtr &` or `const ConstShufflePlugPtr`? I think the former is slightly preferable because it doesn't do any unnecessary manipulation of the reference count for `plug`. And I think that's also what `for( auto &plug )` will choose, so I'd suggest we just use `auto &` here.

I'd really like to address that `\todo` in FilteredChildIterator, but it's a big source compatibility break that might not be easy to swallow for places with significant C++ Gaffer extensions. Maybe I'll do some investigations to see how hard it would hit...

In the meantime, I've updated my local compiler to GCC 11, so I can join you in the quest for VFX2023 readiness...